explainers/signed-exchange-subresource-subtitution-explainer.md

```diff
@@ -0,0 +1,455 @@
+   signed exchange.
+   - After processing the link HTTP headers, clears the
+   PrefetchedSignedExchangeCache.
+
+## Recursive subresource signed exchange prefetch
+
+While prefetching an HTML resource in signed exchange format:
+
+1. When the UA detects a "preload" link HTTP header in the inner response, check
+whether a matching “allowed-alt-sxg” link HTTP header in the inner response
+exists or not. (Note that multiple `allowed-alt-sxg` links can be present for
+the same preload if they include `variants` and `variant-key` attributes. In
+that case, the UA uses the algorithm written in
+[HTTP Representation Variants](https://httpwg.org/http-extensions/draft-ietf-httpbis-variants.html)
+spec to find the matching header.)
+1. If an `allowed-alt-sxg` link exists, check whether the signed exchange was
+served with a matching “alternate” link HTTP header.
+1. If the outer signed exchange did identify an alternate version of the
+subresource, prefetch the subresource signed exchange instead of prefetching the
+original resource URL.
+1. The prefetched signed exchange will be stored to the SignedExchangeCache of
+the Document. And it will be passed to the next Document and used while
+processing the preload link header. This behavior is written in above.
+
+
+# Detailed design discussion
+## Header integrity of signed exchange
+The `header-integrity` parameter of the `allowed-alt-sxg` link is the SHA256
+hash value of the *signedHeaders* value from the
+[application/signed-exchange format](https://wicg.github.io/webpackage/draft-yasskin-http-origin-signed-responses.html#application-signed-exchange)
+for integrity checking. This signedHeaders is *"the canonical serialization of
+the CBOR representation of the response headers of the exchange represented by
+the application/signed-exchange resource, excluding the Signature header
+field"*. So this value doesn’t change even if the publisher signs the content
+again or changes the signing key. This header-integrity value also guarantees
+the content body hasn't changed, because the signed headers are required to
+include a `Digest` header with a hash of the content body.
+
+The UA needs to check that this value of the prefetched subresource signed
+exchange is same as the header-integrity attribute of allowed-alt-sxg link
+header in the response from the publisher. This is intended to prevent the
+distributor from encoding a tracking ID into the subresource signed exchange.
+
+We can’t use the
+[SRI’s integrity](https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity)
+for this purpose, because SRI’s integrity only covers the content body and not
+any of the headers. So if the UA use the SRI’s integrity value in
+allowed-alt-sxg link header, we can use the subresource signed exchanges to
+track the users by changing content-type and detecting the image loading
+failure.
+
+## Multiple subresource signed exchanges
+If there are multiple matching subresource signed exchanges (example:
```

Comment:
It's not clear what "matching" means here. I think you're referring to subresource preloads that are provided by prefetches on the previous page?

Reply:
Changed to
> If there are multiple subresource preloads that are provided by prefetches on
the previous page in signed exchange format